packages/cli/src/commands/model/list.ts

```diff
@@ -0,0 +1,100 @@
+import { Command, type QueryCommandFlags } from '../../command.js'
+import { Flags, CliUx } from '@oclif/core'
+import { Model } from '@ceramicnetwork/stream-model'
+import Table from 'cli-table3'
+import { Edge, Page, PageInfo, StreamState } from '@ceramicnetwork/common'
+import terminalSize from 'term-size'
+
+type PartialModelDefinition = {
+  id: string
+  name: string
+  description: string
+}
+
+type ModelListFlags = QueryCommandFlags & {
+  table?: boolean
+}
+
+export default class ModelList extends Command<ModelListFlags> {
+  fetchedFields: Array<PartialModelDefinition> = []
+  lastLoadedPageInfo: PageInfo | null = null
+
+  static description = 'load a list of models with pagination'
+
+  static flags = {
+    ...Command.flags,
+    table: Flags.boolean({
+      description: 'display the results as a table',
+    }),
+  }
+
+  getPageSize(): number {
+    const rows = terminalSize()['rows']
+    // When we display a table, each row is takes two terminal rows. We subtract additional 3 lines per page, to leave
+    // space for the table header and next page prompt
+    return Math.max(20, this.flags.table ? rows / 2 - 3 : rows)
```

Comment:
Seems this could still lead to a floating point number output if `rows` is an odd number?

Reply:
Ugh, of course, sorry for this, Paul, fixing right now.